test/integration/agent/route_test.go

```diff
@@ -134,40 +132,117 @@ func TestRouteTable(t *testing.T) {
 			t.Errorf("mismatch route tables")
 		}
 
-		if tc.expSvcTbl && tc.podCIDR != nil {
-			expRouteStr := fmt.Sprintf("%s dev %s scope link", tc.podCIDR, gwName)
+		if tc.expSvcTbl {
+			expRouteStr := fmt.Sprintf("%s dev %s scope link", podCIDR, gwName)
 			expRouteStr = strings.Join(strings.Fields(expRouteStr), "")
-			ipRoute, _ := ExecOutputTrim(fmt.Sprintf("ip route show table %d | grep %s", svcTblIdx, tc.podCIDR))
+			ipRoute, _ := ExecOutputTrim(fmt.Sprintf("ip route show table %d | grep %s", svcTblIdx, podCIDR))
 			if len(ipRoute) > len(expRouteStr) {
 				ipRoute = ipRoute[:len(expRouteStr)]
 			}
-			if !assert.Equal(t, expRouteStr, ipRoute) {
-				t.Errorf("mismatch link route")
-			}
+			assert.Equal(t, expRouteStr, ipRoute, "mismatch link route")
 		}
 
 		// verify ip rules
 		expIPRulesStr := ""
 		if tc.expIPRule {
-			expIPRulesStr = fmt.Sprintf("%d: from all fwmark %#x iif %s lookup %s", route.AntreaIPRulePriority, iptables.RtTblSelectorValue,
+			expIPRulesStr = fmt.Sprintf("%d: from all fwmark %#x iif %s lookup %s", route.AntreaIPRulePriority, route.RtTblSelectorValue,
 				gwName, svcTblName)
 			expIPRulesStr = strings.Join(strings.Fields(expIPRulesStr), "")
 		}
-		ipRule, _ := ExecOutputTrim(fmt.Sprintf("ip rule | grep %x", iptables.RtTblSelectorValue))
+		ipRule, _ := ExecOutputTrim(fmt.Sprintf("ip rule | grep %x", route.RtTblSelectorValue))
 		if !assert.Equal(t, expIPRulesStr, ipRule) {
 			t.Errorf("mismatch ip rules")
 		}
 
-		// verify routes
-		var peerCIDR *net.IPNet
-		var nhCIDRIP net.IP
-		if len(tc.peerCIDR) > 0 {
-			_, peerCIDR, _ = net.ParseCIDR(tc.peerCIDR)
-			nhCIDRIP = ip.NextIP(peerCIDR.IP)
+		// verify ipset
```

Comment:
do u want to verify iptables in route test or create a separate test?

Reply:
The iptables logic is encapsulated in route interface in this PR. 
The tests are divided by tested functions, like Initialize which will setup route, ipset, and iptables. Having multiple tests that verify different parts of single function seems not common, I would prefer keeping it common and simple, unless there are good reasons to divide it further.